### PR TITLE
Feature/te 440 use call in assert

### PR DESCRIPTION
**Pull request by gunther-bachmann**
Ambiguities in the grammar had to be resolved:
* expression comparison operator for matching strings (originally 'matches') had to be replaced with an operator '~' (same holds true for the negated version)
* expression comparison operator '<>' had to be dropped 
* the rule that allowed for empty pairs of '<' WS+ '>' (with whitespaces in between) had to be dropped

This pull request is open for discussion, since the added complexity and grammar changes to avoid ambiguities must be worth it ...

### CODE REVIEW

File: tcl/org.testeditor.tcl.dsl.tests/src/org/testeditor/tcl/dsl/jvmmodel/SimpleTclGeneratorIntegrationTest.xtend [hunk 1]
```diff
@@ -66,13 +66,16 @@ class SimpleTclGeneratorIntegrationTest extends AbstractTclGeneratorIntegrationT
 				- assert baz = "fix"
 				- assert book
 				- assert mak."key with spaces" = "fox"
-				- assert ! foo
-				- assert baz <> "fix"
+				- assert false foo
```
Comment (franzbecker) [inline tcl/org.testeditor.tcl.dsl.tests/src/org/testeditor/tcl/dsl/jvmmodel/SimpleTclGeneratorIntegrationTest.xtend]: 😐  so negating an expression is different when we have a variable? That's not very intuitive.

File: tcl/org.testeditor.tcl.dsl/src/org/testeditor/tcl/dsl/Tcl.xtext [hunk 2]
```diff
@@ -183,9 +195,14 @@ StepContentVariable returns tsl::StepContentVariable:
...
-	('<' value=ID '>') |
-	{StepContentElement} ('<>' | '<' WS+ '>');
+	'<' value=ID '>' |
+	{StepContentElement} '<>';
 
 StepContentPunctuation:
-	value=('.'|'?')
-;
\ No newline at end of file
+	DotStepContentPunctuation | BooleanStepContentPunctuation;
```
Comment (franzbecker) [inline tcl/org.testeditor.tcl.dsl/src/org/testeditor/tcl/dsl/Tcl.xtext]: Why do we need this separation?

File: tcl/org.testeditor.tcl.dsl.tests/src/org/testeditor/tcl/dsl/jvmmodel/SimpleTclGeneratorIntegrationTest.xtend (hunk 1 above)
Comment (gunther-bachmann) [inline tcl/org.testeditor.tcl.dsl.tests/src/org/testeditor/tcl/dsl/jvmmodel/SimpleTclGeneratorIntegrationTest.xtend]: Not really. The two lines 'assert ! foo' and 'assert false foo' are equivalent ('false' just an alias for '!' in this case).

**Comment by franzbecker** [comment]
My first impression is that the tradeoff is not worth the benefit of implementing this feature. Apart from the added complexity and the required grammar changes, this might lead to future problems when we want to support the Given/When/Then pattern. Allowing this would mean to explicitly allow the combination of When+Then into a single step.

File: tcl/org.testeditor.tcl.dsl/src/org/testeditor/tcl/dsl/Tcl.xtext (hunk 2 above)
Comment (gunther-bachmann) [inline tcl/org.testeditor.tcl.dsl/src/org/testeditor/tcl/dsl/Tcl.xtext]: I use boolean step content punctuation in one rule to enforce this punctuation for calls within asserts.

**Comment by gunther-bachmann** [comment]
I'm not sure either. But we should let the others think about it, too. Breaking the given/when pattern is IMHO not a problem of this feature itself but of its use. I put a comment on the ticket that looks something like this:

{code}
// when
- Execute some business function

// then
- assert Is <button> enabled?
{code}

This would be a proper usage, wouldn't it?

**Comment by franzbecker** [comment]
We agreed to postpone this feature for now.

**Pull request closed without merging**